noir_stdlib/src/lib.nr

```diff
@@ -121,6 +121,33 @@ where
 #[builtin(as_witness)]
 pub fn as_witness(x: Field) {}
 
+/// A for loop like this:
+///
+/// ```noir
+/// for i in start..min(Max, end) {
+///     f(i);
+/// }
+/// ```
+///
+/// that is implemented using constrained and (an optimized) unconstrained versions.
+pub(crate) fn for_loop<let Max: u32, Env>(start: u32, end: u32, f: fn[Env](u32) -> ()) {
+    if crate::runtime::is_unconstrained() {
+        if end < Max {
```

Comment:
Wouldn't an `assert` be better? I saw it in other cases, and it would catch mistakes at least, while this would silently do nothing. Also, shouldn't it be `end <= Max`? I mean the `..end` is exclusive, isn't it?

Reply:
Good catch, done!